aten/src/ATen/native/AdaptiveAveragePooling.cpp

```diff
@@ -324,12 +329,18 @@ namespace {
     }
 
     if (output_size[0] == 1 && output_size[1] == 1) {
-//in this case, adaptive pooling is just computing mean over hw dimensions, which can be done more efficiently
-       int64_t mean_size = input.size(-1) * input.size(-2);
-       Tensor out = input.contiguous().view({-1, mean_size}).mean(-1);
-       return input.ndimension() == 3 ? out.view({input.size(0), 1, 1}) : out.view({input.size(0), input.size(1), 1, 1});
+      // in this case, adaptive pooling is just computing mean over hw
+      // dimensions, which can be done more efficiently
+      int64_t mean_size = input.size(-1) * input.size(-2);
+      Tensor out = input.contiguous().view({-1, mean_size});
+      if (out.is_quantized()) {
```

Comment:
move it to the if statement (checking input.is_quantized) - you'd skip allocating the output